crates/matrix-sdk/src/room/timeline/event_item.rs

```diff
@@ -166,6 +166,11 @@ impl EventTimelineItem {
         self.is_own
     }
 
+    /// Flag indicating this timeline item can be edited by current user.
+    pub fn is_editable(&self) -> bool {
+        matches!(&self.content, TimelineItemContent::Message(message) if self.is_own() && matches!(message.msgtype(), MessageType::Text(_)))
```

Comment:
Now that I'm thinking more about this, is `Text` really the only messagetype that can be edited? At least emotes should work too, right?

Also this goes over the 100 character line length. Feel free to just turn the outer `matches` into a `match` again, then rustfmt should kick in. Or you can do manual line breaks.

Reply:
Updated to include emotes, reverted to match again. Formatter didn't do anything about the line length btw :/